File: backend/app/schemas/gene.py
from pydantic import BaseModel


class GeneSearchQuery(BaseModel):
    query: str
    organism: str | None = None
    limit: int = 20


class GeneSearchResult(BaseModel):
    gene_id: str
    symbol: str
    description: str
    organism: str
    tax_id: int | None = None


class GeneDetail(BaseModel):
    gene_id: str
    symbol: str
    full_name: str
    description: str
    organism: str
    tax_id: int
    chromosome: str | None = None
    map_location: str | None = None
    aliases: list[str] = []


class GeneSequence(BaseModel):
    gene_id: str
    accession: str
    sequence: str
    sequence_type: str  # "genomic", "mrna", "cds"
    length: int

File: frontend/src/types/gene.ts
export interface GeneSearchResult {
  gene_id: string;
  symbol: string;
  description: string;
  organism: string;
  tax_id?: number;
}

export interface GeneDetail {
  gene_id: string;
  symbol: string;
  full_name: string;
  description: string;
  organism: string;
  tax_id: number;
  chromosome?: string;
  map_location?: string;
  aliases: string[];
}

export interface GeneSequence {
  gene_id: string;
  accession: string;
  sequence: string;
  sequence_type: "genomic" | "mrna" | "cds";
  length: number;
}
